cob_twist_controller/src/damping_methods/damping.cpp

```diff
@@ -76,10 +81,13 @@ inline double DampingNone::getDampingFactor(const Eigen::VectorXd& sorted_singul
 /**
  * Method just returns the damping factor from ros parameter server.
  */
-inline double DampingConstant::getDampingFactor(const Eigen::VectorXd& sorted_singular_values,
+inline Eigen::MatrixXd DampingConstant::getDampingFactor(const Eigen::VectorXd& sorted_singular_values,
                                                 const Eigen::MatrixXd& jacobian_data) const
 {
-    return this->params_.damping_factor;
+    uint32_t rows = sorted_singular_values.rows();
+    Eigen::MatrixXd damping_factor = Eigen::MatrixXd::Zero(rows,rows);
+    damping_factor(sorted_singular_values.rows() - 1,sorted_singular_values.rows() - 1) = pow(this->params_.damping_factor,2);
```

Comment:
In my understanding in the old version we had ```lambda*IdentityMatrix```, which in my understanding results in a DiagonalMatrix(lambda).
Now damping_factor is only a ZeroMatrix with only Index ```(sorted_singular_values.rows() - 1, sorted_singular_values.rows() - 1)``` set to lambda.
This is __different__!
damping_factor should again be a DiagonalMatrix(lambda)!